spec/lexers/csound_spec.rb

```diff
@@ -0,0 +1,425 @@
+        ['Punctuation', "'"],
+        ['Text', ' '],
+        ['Comment.Preproc', 'ARG3'],
+        ['Punctuation', ')'],
+        ['Comment.Multiline', '/**/'],
+        ['Text', "\n"],
+        ['Punctuation', '#'],
+        ['Comment.Preproc', '\\#'],
+        ['Name', 'macro'],
+        ['Text', "\n"],
+        ['Name', 'body'],
+        ['Comment.Preproc', '\\#'],
+        ['Punctuation', '#'],
+        ['Text', "\n"]
+    end
+
+    it 'lexes macro preprocessor directives' do
+      %w(#ifdef #ifndef #undef).each do |directive|
+        assert_tokens_equal "#{directive} MACRO",
+          ['Comment.Preproc', directive],
+          ['Text', ' '],
+          ['Comment.Preproc', 'MACRO']
+      end
+    end
+
+    it 'lexes other preprocessor directives' do
+      code = <<-EOF.unindent
+        #else
+        #end
+        #endif
+        ###
+        @ \t12345
+        @@ \t67890
+      EOF
+      assert_tokens_equal code,
+        ['Comment.Preproc', '#else'],
+        ['Text', "\n"],
+        ['Comment.Preproc', '#end'],
+        ['Text', "\n"],
+        ['Comment.Preproc', '#endif'],
+        ['Text', "\n"],
+        ['Comment.Preproc', '###'],
+        ['Text', "\n"],
+        ['Comment.Preproc', "@ \t12345"],
+        ['Text', "\n"],
+        ['Comment.Preproc', "@@ \t67890"],
+        ['Text', "\n"]
+    end
+
+    it 'lexes function-like macros' do
+      assert_tokens_equal %q[$MACRO.(((x\\))' "x)\\)x\\))"# {{x\\))x)\\)}})],
+        ['Comment.Preproc', '$MACRO.'],
+        ['Punctuation', '('],
+        ['Comment.Preproc', '(('],
+        ['Name', 'x'],
+        ['Comment.Preproc', '\\)'],
+        ['Error', ')'],
+        ['Punctuation', "'"],
+        ['Text', ' '],
+        ['Literal.String', '"x'],
+        ['Error', ')'],
+        ['Comment.Preproc', '\\)'],
+        ['Literal.String', 'x'],
+        ['Comment.Preproc', '\\)'],
+        ['Error', ')'],
+        ['Literal.String', '"'],
+        ['Punctuation', '#'],
+        ['Text', ' '],
+        ['Literal.String', '{{x'],
+        ['Comment.Preproc', '\\)'],
+        ['Error', ')'],
+        ['Literal.String', 'x'],
+        ['Error', ')'],
+        ['Comment.Preproc', '\\)'],
+        ['Literal.String', '}}'],
+        ['Punctuation', ')']
+    end
+
+    include Support::Guessing
+    it 'guesses by filename' do
+      assert_guess :filename => 'foo.orc'
+      assert_guess :filename => 'foo.udo'
+    end
+  end
+end
+
+
+describe Rouge::Lexers::CsoundScore do
+  let(:subject) { Rouge::Lexers::CsoundScore.new }
+
+  describe 'Csound Score lexer' do
+    include Support::Guessing
+    it 'guesses by filename' do
+      assert_guess :filename => 'foo.sco'
+    end
+  end
+end
+
+
+describe Rouge::Lexers::CsoundDocument do
+  let(:subject) { Rouge::Lexers::CsoundDocument.new }
+
+  describe 'Csound Document lexer' do
+    include Support::Guessing
+    it 'guesses by filename' do
+      assert_guess :filename => 'foo.csd'
+    end
+  end
+end
```

Comment:
Sorry, I’m afraid I still don’t understand. Of the 22 specs in [spec/lexers/csound_spec.rb](https://github.com/nwhetsell/rouge/blob/master/spec/lexers/csound_spec.rb), [19](https://github.com/nwhetsell/rouge/blob/master/spec/lexers/csound_spec.rb#L20-L393) use [`assert_tokens_equal`](https://github.com/jneen/rouge/search?q=assert_tokens_equal+path%3Aspec%2Fsupport+filename%3Alexing.rb). If removing the 19 specs that use `assert_tokens_equal` is a requirement to get this merged, then I truly appreciate your attention to this pull request, and I think it makes sense to close it without merging, as I’m unable to maintain a lexer that lacks tests for the tokens it emits. It may also be worth considering deprecating `assert_tokens_equal`.